test/test_nn.py

```diff
@@ -4821,6 +4821,20 @@ def test_pairwise_distance(self):
         input2 = torch.randn(4, 4, requires_grad=True)
         self.assertTrue(gradcheck(lambda x, y: F.pairwise_distance(x, y), (input1, input2)))
 
+    def test_pdist(self):
+        for trans in [False, True]:
+            inp = torch.randn(4, 5, requires_grad=True)
+            if trans:
+                inp = inp.transpose(0, 1)
+            for p in [0, 1, 2, 0.5, 1.5, 2.5, float('inf')]:
+                self.assertTrue(gradcheck(lambda x: F.pdist(x, p), (inp,)))
+
+    def test_pdist_zeros(self):
+        for trans in [False, True]:
+            inp = torch.randn(1, 3, requires_grad=True).repeat([2, 1])
+            for p in [0, 1, 2, 0.5, 1.5, 2.5, float('inf')]:
+                self.assertTrue(gradcheck(lambda x: F.pdist(x, p), (inp,)))
```

Comment:
nit: In situations like this I find it helpful to throw in an expected failure case when gradgradcheck is called.  This means if someone fixes double backwards they have to remember to test it.